ksql-engine/src/test/resources/query-validation-tests/between.json

```diff
@@ -0,0 +1,165 @@
+{
+  "comments": [
```

Comment:
consider adding test cases to cover _unsupported_ operations, e.g. I'm assuming it doesn't support function calls in the bound expressions. The test can ensure the error message is as expected. (You can define `expectedException`).

Reply:
For now I'll leave the tests in CodeGenRunnerTest - I can open a new PR with more tests if you feel strongly about adding them to the JSON suite (which tend to run a little longer, which is why I'm erring toward unit tests)